overlord/state/task.go

```diff
@@ -237,9 +236,9 @@ func (t *Task) SetStatus(new Status) {
 	t.changeStatus(old, new)
 }
 
-// Wait puts the task into WaitStatus, and sets the status the task should be restored
-// to after the Wait.
-func (t *Task) Wait(resultStatus Status) {
+// SetToWait puts the task into WaitStatus, and sets the status the task should be restored
+// to after the SetToWait.
+func (t *Task) SetToWait(resultStatus Status) {
```

Comment:
we should probably panic here if resultStatus is DefaultStatus or WaitStatus itself?

Reply:
Added a panic for those!